app/src/main/java/de/danoeh/antennapod/activity/gpoddernet/GpodnetAuthenticationActivity.java

```diff
@@ -123,53 +125,58 @@ public void onClick(View v) {
                 final String usernameStr = username.getText().toString();
                 final String passwordStr = password.getText().toString();
 
-                if (BuildConfig.DEBUG) Log.d(TAG, "Checking login credentials");
-                AsyncTask<GpodnetService, Void, Void> authTask = new AsyncTask<GpodnetService, Void, Void>() {
+                if(usernameHasUnwantedChars(usernameStr)){
+                    txtvError.setText(R.string.gpodnetsync_username_characters_error);
+                    txtvError.setVisibility(View.VISIBLE);
```

Comment:
I think if you add `return` here, you don't need to add the else block. You can just leave the second half of the function as it currently is. That makes it easier to read because there are less indentations.

Please also make sure to clear the error when correcting the text and clicking again.

Reply:
Let me work on this.

Reply:
@ByteHamster Thanks for the feedback. i have implemented it as suggested. As for the clearing the text when the user enters the an acceptable string into the `username` field the `aysnctask` will be executed and in its `onPreExecute` method the the `txtvError` textview  visibility will be set to gone. If the user inputs a string that has unacceptable characters the message  will be displayed.